hkmc2/shared/src/test/mlscript-compile/Runtime.mjs

```diff
@@ -11,52 +14,66 @@ let Runtime1;
       toString() {
         return "()"
       }
+      static [constructorName]() { return "Unit"; }
+      static [definitionKind]() { return "object"; }
     };
     this.Unit = new Unit$class;
-    this.Unit.class = Unit$class;
+    Object.defineProperty(this.Unit, 'class', { value: Unit$class });
     this.try_catch = RuntimeJS.try_catch;
     this.EffectHandle = function EffectHandle(_reified1) {
       return new EffectHandle.class(_reified1);
     };
-    this.EffectHandle.class = class EffectHandle {
-      #_reified;
-      constructor(_reified) {
-        this.#_reified = _reified;
-        this.reified = this.#_reified;
-      }
-      resumeWith(value) {
-        let lambda;
-        const this$EffectHandle = this;
-        lambda = (undefined, function () {
-          let tmp;
-          tmp = Runtime.resume(this$EffectHandle.reified.contTrace);
-          return runtime.safeCall(tmp(value))
-        });
-        return Runtime1.try(lambda)
-      } 
-      raise() {
-        return Runtime.topLevelEffect(this.reified, false)
+    Object.defineProperty(this.EffectHandle, 'class', {
+    value: class EffectHandle {
+        #_reified;
+        constructor(_reified) {
+          this.#_reified = _reified;
+          this.reified = this.#_reified;
+        }
+        resumeWith(value) {
+          let lambda;
+          const this$EffectHandle = this;
+          lambda = (undefined, function () {
+            let tmp;
+            tmp = Runtime.resume(this$EffectHandle.reified.contTrace);
+            return runtime.safeCall(tmp(value))
+          });
+          return Runtime1.try(lambda)
+        } 
+        raise() {
+          return Runtime.topLevelEffect(this.reified, false)
+        }
+        static [constructorName]() { return "EffectHandle"; }
+        static [fieldNames]() { return []; }
+        static [definitionKind]() { return "class"; }
       }
-      toString() { return "EffectHandle(" + "" + ")"; }
-    };
+    });
     this.MatchResult = function MatchResult(captures1) {
       return new MatchResult.class(captures1);
     };
-    this.MatchResult.class = class MatchResult {
-      constructor(captures) {
-        this.captures = captures;
+    Object.defineProperty(this.MatchResult, 'class', {
+    value: class MatchResult {
+        constructor(captures) {
+          this.captures = captures;
+        }
+        static [constructorName]() { return "MatchResult"; }
+        static [fieldNames]() { return ["captures"]; }
+        static [definitionKind]() { return "class"; }
```

Comment:
As discussed, to avoid needless allocations and lookups, this should be something like
```suggestion
        static [constructorInfo] = return ["class", "MatchResult", "captures"]
```
Is there any reason to make it a function?

Reply:
Ok, fine.